src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -311,6 +311,8 @@ protected void AddCoins(TransactionBuildContext context)
             if (balance < totalToSend)
                 throw new WalletException("Not enough funds.");
 
+            bool onlySelectedInputs = false;
```

Comment:
maybe have a variable like so:
`bool onlySelectedInputs = context.SelectedInputs != null && context.SelectedInputs.Any() && !context.AllowOtherInputs`;
It's more long-winded but I think it's better than modifying flags when we reach some parts of the code.

Reply:
See previous comment.